### ISSUE TITLE
No longer forcing deep updates?

### ISSUE THREAD

**Issue by syranide** [gaearon/react-hot-loader#83]
PR: gaearon/react-hot-api#7

----

**Root.jsx**
```
var React = require('react');
var Child = require('./Child');

class Root extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <div>
        ROOT
        <Child />
      </div>
    );
  }
}

module.exports = Root;
```

**Child.jsx**
```
var React = require('react');

class Child extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <div>
        CHILD
      </div>
    );
  }
}

module.exports = Child;
```

Changes to Root.jsx show up, but changes to Child.jsx don't. Did something break? (PS. React 0.13)

cc @gaearon

**Comment by gaearon** [comment]
Thanks! Does this happen with 0.13.0-beta.1 or master?

**Comment by syranide** [comment]
`npm install react@0.13` gives you `react@0.13.0-beta.2` which is what I'm using.

**Comment by gaearon** [comment]
Oh, I see. I didn't realize there's a new beta. I'll check it out later today.

**Comment by mindjuice** [comment]
I'm seeing something similar.  I have one a really simple set of components as I'm just getting setup for a new project.  The App component renders three child components, Inbox, Calendar and Dashboard.  The child components are identical except for the text displayed in them.  Oddly though, changing App, Inbox or Calendar DOES trigger a reload, but changing Dashboard does not.

I'm on React 0.12.2 and react-hot-loader 1.1.5.

**Comment by syranide** [comment]
PR merged, this is fixed in next release.

**Comment by mindjuice** [comment]
In my case, I just hadn't properly configured react-hot-loader.  It's working for me now.  Such a newb!

**Comment by gaearon** [comment]
Fixed in `react-hot-api@0.3.5`, which should be installed when doing `npm install`!
